third_party/move/move-vm/types/src/utils/union_find.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Debug)]
+pub struct UnionFind {
+    parent: Vec<u32>,
+    rank: Vec<u8>,
+    num_sets: usize,
+    capacity: usize,
+}
+
+impl UnionFind {
+    /// Creates a new UnionFind instance with `num_elements` and `capacity`
```

Comment:
nit: you can add `[..]` around type name to make it searchable in IDEs, and for refactorings (e.g., if you refactor via RustRover the comment will stay with the old name):
```rust
/// Creates a new [UnionFind] ...
```
Let's use this? Applies to all comments below as well. Also I am kind of not a fan of comments like `..."num_elements" and "capacity"...` because these can change names in the signature. Can we just say: "with the specified number of elements and capacity"?